torch/serialization.py

```diff
@@ -453,14 +453,20 @@ def persistent_load(saved_id):
             raise RuntimeError("Unknown saved id type: %s" % saved_id[0])
 
     f_is_real_file = _is_real_file(f)
-    if f_is_real_file and f.tell() == 0:
-        # legacy_load requires that f has fileno()
-        # only if offset is zero we can attempt the legacy tar file loader
-        try:
-            return legacy_load(f)
-        except tarfile.TarError:
-            # if not a tarfile, reset file offset and proceed
-            f.seek(0)
+
+    try:
+        if f_is_real_file and f.tell() == 0:
```

Comment:
This works, but it doesn't catch the following case:
Let's say f is not backed by a real file descriptor, but is instead a python file-like object. I'm not sure if we still expect `f` to be seekable, but if we try seeking on it, the error message might still be weird.

Reply:
Will the line above `f_is_real_file = _is_real_file(f)` mitigate this issue? Or you mean the file-like object can still pass the `_is_real_file()` check? If that's the case, then we can probably improve the check.